src/main/scala/cromwell/engine/backend/jes/Run.scala

```diff
@@ -29,60 +27,68 @@ object Run  {
     rpr.setLogging(logging)
 
     val id = pipeline.genomicsService.pipelines().run(rpr).execute().getName
-    Log.info(s"$tag ID is $id")
-    new Run(id, pipeline)
+    Log.info(s"$tag JES ID is $id")
+    new Run(id, pipeline, tag)
   }
 
   implicit class RunJesParameters(val params: Seq[JesParameter]) extends AnyVal {
     def toRunMap = params.map(p => p.name -> p.gcs).toMap.asJava
   }
 
+  implicit class RunOperationExtension(val operation: Operation) extends AnyVal {
+    def hasStarted = operation.getMetadata.asScala.get("startTime") isDefined
+  }
+
   sealed trait RunStatus
   trait TerminalRunStatus extends RunStatus
-  final case class Initializing(created: DateTime) extends RunStatus
-  final case class Running(created: DateTime, started: DateTime) extends RunStatus
-  final case class Success(created: DateTime, started: DateTime, finished: DateTime) extends TerminalRunStatus
-  final case class Failed(created: DateTime, started: DateTime, finished: DateTime, errorCode: Int, errorMessage: String) extends TerminalRunStatus
+  case object Initializing extends RunStatus
+  case object Running extends RunStatus
+  case object Success extends TerminalRunStatus
+  final case class Failed(errorCode: Int, errorMessage: String) extends TerminalRunStatus {
+    // Don't want to include errorMessage or code in the snappy status toString:
+    override def toString = "Failed"
+  }
 }
 
-case class Run(name: String, pipeline: Pipeline) {
+case class Run(name: String, pipeline: Pipeline, tag: String) {
+
   def status(): RunStatus = {
     val op = pipeline.genomicsService.operations().get(name).execute
-    val metadata = OperationMetadata(op)
 
     if (op.getDone) {
-      val error = Option(op.getError)
-      error match {
-        case Some(x) => Failed(
-          metadata.created,
-          metadata.started.get,
-          metadata.finished.get,
-          x.getCode,
-          x.getMessage)
-        case None => Success(metadata.created, metadata.started.get, metadata.finished.get)
-      }
-    } else metadata.started map {s => Running(metadata.created, s)} getOrElse Initializing(metadata.created)
+      // If there's an error, generate a Failed status. Otherwise, we were successful!
+      Option(op.getError) map { x => Failed(x.getCode, x.getMessage) } getOrElse Success
+    } else if (op.hasStarted) {
+      Running
+    } else {
+      Initializing
+    }
   }
 
   @tailrec
-  final def waitUntilComplete(): TerminalRunStatus = {
+  final def waitUntilComplete(previousStatus: Option[RunStatus]): TerminalRunStatus = {
     val currentStatus = status()
-    println(s"Current status is $currentStatus")
+
+    val prevStateName = previousStatus map { _.toString } getOrElse previousStatus.toString
```

Comment:
Having watched the evolution of this code I know what it does and that it's intentional, but IMHO it could use a comment as the `toString`ing of `None` is a little surprising.